src/syntax/Parser.js

```diff
@@ -389,11 +431,24 @@ export class Parser {
   parseScript() {
     this.strictMode_ = false;
     let start = this.getTreeStartLocation_();
+    let fs = this.enterFunctionState_(FUNCTION_STATE_SCRIPT);
     let scriptItemList = this.parseStatementList_(true);
     this.eat_(END_OF_FILE);
+    this.leaveFunctionState_(fs);
```

Comment:
This pattern could be changed to not pass in `fs` here. That was useful when developing this to catch potential errors.